src/scripts/Sitemap.js

```diff
@@ -174,26 +174,39 @@ export default class Sitemap {
 		}
 	}
 
-	deleteSelector(selectorToDelete) {
-		this.selectors.forEach(
-			function (selector) {
-				if (selector.hasParentSelector(selectorToDelete.uuid)) {
-					selector.removeParentSelector(selectorToDelete.uuid);
-					if (selector.parentSelectors.length === 0) {
-						this.deleteSelector(selector);
-					}
-				}
-			}.bind(this)
-		);
+	redundantParentsCleaner(selectorsList) {
+		const currentUuidList = selectorsList.map(selector => selector.uuid);
+		currentUuidList.push(this.rootSelector.uuid);
+		selectorsList.forEach(selector => {
+			selector.parentSelectors = selector.parentSelectors.filter(uuid =>
+				currentUuidList.includes(uuid)
+			);
+		});
+		return selectorsList;
+	}
+	createRemainingSelectorsList(parentSelector) {
+		let newList = [];
+		let selectorsQueue = [this.rootSelector];
+		while (selectorsQueue.length > 0) {
+			let currentSelector = selectorsQueue.shift();
+			const selectorChilds = this.selectors.filter(selector =>
+				selector.parentSelectors.includes(currentSelector.uuid)
+			);
 
-		for (const i in this.selectors) {
-			if (this.selectors[i].uuid === selectorToDelete.uuid) {
-				this.selectors.splice(i, 1);
-				break;
-			}
+			selectorChilds.forEach(function (child) {
+				if (child.uuid !== parentSelector.uuid) {
```

Comment:
Here you should check if you have seen child selector already

Also some minor things
- use const for variables
- use arrow function here
- rename `selectorChilds` to `selectorChildren`
- it is not clear from `createRemainingSelectorsList(parentSelector)` signature that you are collecting remaining selectors after deletion; I suggest you could at least rename the argument to `selectorToDelete`
- rename `redundantParentsCleaner` to `cleanRedundantParens` - prefer verb phrases for methods names
- ensure there are newlines between methods